tables/mapaction_data_table.sql

```diff
@@ -821,6 +821,100 @@ where
     tags @> '{"amenity":"water_point"}';
 
+-- 
+insert into :ma_table(ma_category, ma_theme, ma_tag, fclass, feature_type, geom, osm_minimum_tags, osm_id, osm_type)
+select 'admn',
+    'ad0',
+    'adminboundary0',
+    tags ->> 'admin_level',
+    'py',
+    geog::geometry as geom,
+    tags,
+    osm_id,
+    osm_type
+from :osm_table
+where tags @> '{"boundary":"administrative"}'
+    and tags @> '{"admin_level":"2"}'
+    and geometrytype(geog::geometry) ~* 'polygon'
+    and tags ->> 'admin_level' ~ '^\d+$';
+
+-- 
+insert into :ma_table(ma_category, ma_theme, ma_tag, fclass, feature_type, geom, osm_minimum_tags, osm_id, osm_type)
+select 'admn',
+    'ad1',
+    'adminboundary1',
+    tags ->> 'admin_level',
+    'py',
+    geog::geometry as geom,
+    tags,
+    osm_id,
+    osm_type
+from :osm_table
+where tags @> '{"boundary":"administrative"}'
+    and geometrytype(geog::geometry) ~* 'polygon'
+    and tags ->> 'admin_level' ~ '^\d+$'
+    and tags @> format('{"admin_level":%1$I}', (select distinct (tags ->> 'admin_level')::int
+        from :osm_table
+        where  tags @> '{"boundary":"administrative"}'
+            and tags ? 'admin_level'
+            and geometrytype(geog::geometry) ~* 'polygon'
+            and tags ->> 'admin_level' ~ '^\d+$'
+        order by 1
+        offset 1
+        limit 1
+    ))::jsonb
+
+-- 
+insert into :ma_table(ma_category, ma_theme, ma_tag, fclass, feature_type, geom, osm_minimum_tags, osm_id, osm_type)
+select 'admn',
+    'ad2',
+    'adminboundary2',
+    tags ->> 'admin_level',
+    'py',
+    geog::geometry as geom,
+    tags,
+    osm_id,
+    osm_type
+from :osm_table
+where tags @> '{"boundary":"administrative"}'
+    and geometrytype(geog::geometry) ~* 'polygon'
+    and tags ->> 'admin_level' ~ '^\d+$'
+    and tags @> format('{"admin_level":%1$I}', (select distinct (tags ->> 'admin_level')::int
+        from :osm_table
+        where  tags @> '{"boundary":"administrative"}'
+            and tags ? 'admin_level'
+            and geometrytype(geog::geometry) ~* 'polygon'
+            and tags ->> 'admin_level' ~ '^\d+$'
+        order by 1
+        offset 2
+        limit 1
+    ))::jsonb
+
+-- 
+insert into :ma_table(ma_category, ma_theme, ma_tag, fclass, feature_type, geom, osm_minimum_tags, osm_id, osm_type)
+select 'admn',
+    'ad3',
+    'adminboundary3',
+    tags ->> 'admin_level',
+    'py',
+    geog::geometry as geom,
+    tags,
+    osm_id,
+    osm_type
+from :osm_table
+where tags @> '{"boundary":"administrative"}'
+    and geometrytype(geog::geometry) ~* 'polygon'
+    and tags ->> 'admin_level' ~ '^\d+$'
+    and tags ->> 'admin_level' in (select distinct tags ->> 'admin_level'
+        from :osm_table
+        where  tags @> '{"boundary":"administrative"}'
+            and tags ? 'admin_level'
+            and geometrytype(geog::geometry) ~* 'polygon'
+            and tags ->> 'admin_level' ~ '^\d+$'
+        order by 1
+        offset 3
```

Comment:
why `offset 3` is required here?

Reply:
because ad0=2 is starting point
ad1 is next step after 2, it can be 3 or 4 = offset 1
ad2 is next step after ad1, it can be 4,5,6 = offset 2
ad3 everything else = offset 3

Reply:
please add comment in the code